### ISSUE TITLE
Time out error when adding AC1715/70 (fix is in last comment)

### ISSUE THREAD

**Issue by nashnix** [kongo09/philips-airpurifier-coap#88]
Hi,

I've connected my AC1715/70 to my wifi, assigned it a static IP, and it is talking happily to the network and the app, but not HomeAssistant; it returns a "Timeout, cannot connect to device" error every time. This is a brand-new 1715 running WiFi Firmware 73.1 and Device Firmware: 0.9.0. I have tried power cycling it (several times) with no luck.

**Related issues:**
- #80 - my AC2939 wasn't working until I made the fix suggested by vszurma (extending the time to failure from 30s to 180s) and now it works without a problem;
- #42 (closed) - This was the original discussion about supporting the 1715 series - that seems to have been focused on the 1715/11;
- #79 (closed) - where I learned I wasn't running commands properly while checking out another time out issue.

Since yesterday, I've managed to wrangle the virtual machine into cooperating and providing log outputs:

**Log output - turning it on:**

> (env) homeassistant:/config# aioairctrl --host XXX.XXX.XXX.XXX status --json
> _WARNING:coap:Did not receive requested pktinfo ancdata on message from ('::ffff:XXX.XXX.XXX.XXX', 5683, 0, 0)_
> _WARNING:coap:Did not receive requested pktinfo ancdata on message from ('::ffff:XXX.XXX.XXX.XXX', 5683, 0, 0)_
> {"D01-02": "Purifier product", "D01-03": "AP1715", "D01-04": "Nala", "D01-05": "AC1715/70", "D01-07": "English", "D01-08": 2, "D01-09": 1, "D01-10": 7, "D01-11": 0, "D01-20": "0.0.1", "D01-21": "0.9.0", "MCUBoot": true, "Runtime": 2884150, "rssi": -32, "otacheck": false, "wifilog": false, "free_memory": 56704, "WifiVersion": "AWS_Philips_AIR@73.1", "ProductId": "bf663606179d11ec8e3306d016384e4a", "DeviceId": "57d2b1a0371411eeae728fda2f90d922", "StatusType": "control", "ConnectType": "Online", **"D03-02": "ON",** "D03-03": false, "D03-04": 100, "D03-05": 100, "D03-11": "A", "D03-12": "Auto General", **"D03-13": "0",** "D03-32": 1, "D03-33": 4, "D03-42": "PM2.5", "D03-43": "PM2.5", "D03-44": 4, "aqit_ext": 0, "D03-64": 0, "D05-02": "A3", "D05-03": "none", "D05-07": 720, "D05-08": 4800, "D05-09": 0, "D05-13": 719, "D05-14": 4799, "D05-15": 0}

**Log output - turning it off:**

> (env) homeassistant:/config# aioairctrl --host XXX.XXX.XXX.XXX status --json
> _WARNING:coap:Did not receive requested pktinfo ancdata on message from ('::ffff:XXX.XXX.XXX.XXX', 5683, 0, 0)_
> _WARNING:coap:Did not receive requested pktinfo ancdata on message from ('::ffff:XXX.XXX.XXX.XXX', 5683, 0, 0)_
> {"D01-02": "Purifier product", "D01-03": "AP1715", "D01-04": "Nala", "D01-05": "AC1715/70", "D01-07": "English", "D01-08": 2, "D01-09": 1, "D01-10": 7, "D01-11": 0, "D01-20": "0.0.1", "D01-21": "0.9.0", "MCUBoot": true, "Runtime": 2889008, "rssi": -32, "otacheck": false, "wifilog": false, "free_memory": 56704, "WifiVersion": "AWS_Philips_AIR@73.1", "ProductId": "bf663606179d11ec8e3306d016384e4a", "DeviceId": "57d2b1a0371411eeae728fda2f90d922", "StatusType": "control", "ConnectType": "Online", **"D03-02": "OFF",** "D03-03": false, "D03-04": 100, "D03-05": 100, "D03-11": "A", "D03-12": "Auto General", **"D03-13": "1",** "D03-32": 1, "D03-33": 4, "D03-42": "PM2.5", "D03-43": "PM2.5", "D03-44": 4, "aqit_ext": 0, "D03-64": 0, "D05-02": "A3", "D05-03": "none", "D05-07": 720, "D05-08": 4800, "D05-09": 0, "D05-13": 719, "D05-14": 4799, "D05-15": 0}

Where it says "WARNING:coap:Did not receive requested pktinfo ancdata on message from ('::ffff:XXX.XXX.XXX.XXX', 5683, 0, 0)" - that was between when I started logging, and when I sent the instructions. It wasn't collecting anything then because there was, at that point, nothing to collect (I was going to delete it, but thought that's bad practice, so explained instead).

The differences between the two entries are:
- D03-02 changed from "ON" to "OFF" when turned off (and from "OFF" to "ON" when turned on)
- D03-13 changed from "0" to "1" when turned off (and from "1" to "0" when turned on)

From these log entries, I've compiled the table below:

| Command sent in the Philips App | Change **FROM** | Change **TO** |
|---|---|---|
|Device turn on| "D03-02": "OFF"<br>"D03-13": "1" | "D03-02": "ON"<br>"D03-13": "0" |
|Device turn off| "D03-02": "ON"<br>"D03-13": "0" |"D03-02": "OFF"<br>"D03-13": "1" |
|Mode set to "Sleep" | "D03-12": "Auto General"<br>"D03-13": "0"  | "D03-12": "Sleep"<br>"D03-13": "1" |
|Mode set to "Turbo" | "D03-12": "Sleep"<br>"D03-13": "0" | "D03-12": "Turbo"<br>"D03-13": "s" |
|Mode set to "1" | "D03-12": "Turbo"<br>"D03-13": "s" | "D03-12": "Gentle/Speed 1"<br>"D03-13": "t" |
|Mode set to "2" | "D03-12": "Gentle/Speed 1"<br>"D03-13": "t" | "D03-12": "Speed 2"<br>"D03-13": "1" |
|Mode set to "Auto" | "D03-12": "Speed 2"<br>"D03-13": "1" | "D03-12": "Auto General"<br>"D03-13": "s" |
|Light panel turned on | "D03-04": "0"<br>"D03-05": 0 | "D03-04": "100"<br>"D03-05": 100
|Light panel turned off | "D03-04": "100"<br>"D03-05": 100 | "D03-04": "0"<br>"D03-05": 0

When I was changing modes I just set it to the next one in the sequence down the menu until it was time to go back to Auto. The light panel has only got on/off options, and just controls the lights on top. 

I appreciate you're very busy with your day job at the moment kongo09, and it might be a while before you can investigate - thank you in any case.

**Comment by nashnix** [comment]
@kongo09 I have good news and strange news.

The good news is; both the 1715 and the 2939 are - as of about five minutes ago - working flawlessly with HomeAssistant, and are even able to translate across into HomeKit.

The strange news is; HomeAssistant refused to believe either of them existed until I did [what eavyon did under Issue 82](https://github.com/kongo09/philips-airpurifier-coap/issues/82), and added both devices to the AirMatters app (no registration of an account or anything like that) and suddenly they were visible and working with HA. I also ratched up the timeout time to 180s as per the fix at issue 80 noted above, but both devices were seen and reporting within seconds.

Unlike eavyon I didn't tell my firewall to block either air purifier from the internet (my workaround until this was functional involved dummy switches, shortcuts, a program running on a mac called "Signals", and the official Philips Air+ app - and for that, they needed to see the internet). It's more like that they were refusing to talk to anyone but the Philips app until AirMatters came knocking, and then suddenly they were visible to your integration as well.

I won't close this issue just yet though; just because it *started* working doesn't mean it will continue to do so, and I'd like to wait at least 48 hours before saying it's all good. If everything is still running happily come Monday (Aussie Time), then I'll be happy to call this one resolved.

**Comment by nashnix** [comment]
TL/DR for @kongo09 - it's working now, no need for you to do anything. The fix is below for anyone who has the same problem with the 1715 (and also if the 2939 starts having tantrums).

------

Five days later, and I can report that both devices are happily and consistently communicating with the plugin. If anyone else ever encounters this issue, here's the steps I did that fixed it:

1. Install this plugin (do not try to connect just yet);
2. Edit the config files of this plugin so that the timeout is at 180s (see the link in my first post for how and where and so on);
3. Get the AirMatters app - **do not** bother registering an account or anything (make sure your phone is on the same network as HomeAssisstant and the purifiers);
4. Now tell this plugin to try to connect to your purifiers - you can do it via IP address or just let it see what it can see;
5. **While it is trying to connect** open the AirMatters app and connect *that* to each purifier you want to hook up to the integration; and
6. Add each purifier to the integration.

And it ought to work quite happily; no need to power cycle the purifiers at all or reboot HA or your network. For some reason using the AirMatters app informs/reminds the purifiers that there's a whole local network they can talk to and get instructions from. Once you do that, the integration has zero issues talking to the purifiers.

A few other points:

- IPv6 should to be enabled on your network - I checked, and turning that off caused _both_ purifiers to have a panic attack about the world in general.
- If you restart HomeAssistant, the odds are pretty high that the plugin will refuse to see one or both purifiers (there's no telling which one - only once was the 1715 the problem child here; every other time it's been the 2939) - to fix this, just open the AirMatters app, and wait until it sees the purifiers. Once it does, reload the configuration of the device on HA that is having issues (under the integration, click on the three dots and select "reload").
- Related; if the connection between this integration and a purifier ever drops and you haven't restarted HA, open the AirMatters app, then reload the configuration of the device on HA.

---

**Subsequent Edit**

When restarting HomeAssistant for any reason do the following:

- FIRST, disable all devices in the integration *before* restarting (do not disable the integration itself);
- Have AirMatters open on your phone (not necessary, but it helps);
- Restart HA as normal; then
- Once HA is up and running, re-enable each device one at a time, waiting for each one to connect and register before re-enabling the next one.

If you do this, the connections happen smoothly and without a problem, and HA boots up *much* faster too.

This has been tested on both models noted at the start, and is working perfectly for both.
